src/node.rs

```diff
@@ -240,9 +241,8 @@ where
     /// Constructs a new bin from the given nodes.
     ///
     /// Nodes are arranged into an ordered red-black tree.
-    pub(crate) fn new(bin: Owned<BinEntry<K, V>>, guard: &Guard) -> Self {
+    pub(crate) fn new(bin: Shared<'_, BinEntry<K, V>>, guard: &Guard<'_>) -> Self {
```

Comment:
The intention here was that you can only construct a `Node` from a `bin` you _know_ is initially owned. This new interface _allows_ someone to call `Node::new` with a thing that's _already_ shared, which I'm not sure is safe.

Reply:
Maybe it could just take a `BinEntry` directly and do the `Shared::new` bit internally?

Reply:
Sure.

Reply:
Hm, it seems the bin might be null?

Reply:
I'm not sure I follow?

Reply:
Misunderstood how table bins were created, fixed in 386eaa75f33a0aab33646242525b45af5adf26ac.